enterprise/api_client/enterprise.py

```diff
@@ -28,20 +29,57 @@ class EnterpriseApiClient(JwtLmsApiClient):
 
     DEFAULT_VALUE_SAFEGUARD = object()
 
-    def get_enterprise_courses(self, enterprise_customer, **kwargs):
+    def get_enterprise_courses(self, enterprise_customer):
         """
-        Query the Enterprise API for the course details of the given course_id.
+        Query the Discovery API for the course details of the given all courses present in enterprise catalogs.
+
         Arguments:
-            enterprise_customer (Enterprise Customer): Enterprise customer for fetching courses
+            enterprise_customer (Enterprise Customer): Enterprise customer for fetching courses.
+
         Returns:
             dict: A dictionary containing details about the course, in an enrollment context (allowed modes, etc.)
         """
-        return self._load_data(
-            self.ENTERPRISE_CUSTOMER_ENDPOINT,
-            detail_resource='courses',
-            resource_id=str(enterprise_customer.uuid),
-            **kwargs
-        )
+        discovery_api = CourseCatalogApiServiceClient(enterprise_customer.site)
+
+        response = {
+            'count': 0,
+            'next': None,
+            'previous': None,
+            'results': []
+        }
+
+        if enterprise_customer.catalog:
+            response = self._load_data(
+                self.ENTERPRISE_CUSTOMER_ENDPOINT,
+                detail_resource='courses',
+                resource_id=str(enterprise_customer.uuid),
+                traverse_pagination=True,
+            )
+
+        for enterprise_customer_catalog in enterprise_customer.enterprise_customer_catalogs.all():
+            catalog_courses = discovery_api.get_search_results(enterprise_customer_catalog.content_filter)
```

Comment:
Scratch that. This is the Enterprise API client. We should be calling the Enterprise API's `enterprise_customer_catalogs` endpoint here to retrieve the course run data. This will ensure that the course enrollment URL is populated.